adapters/ix/ix_test.go

```diff
@@ -1,643 +1,19 @@
-	ctx := context.TODO()
-	pbReq := pbs.PBSRequest{}
-	adUnit1 := getAdUnit()
-	adUnit2 := getAdUnit()
-	adUnit2.Code = "unitCode2"
-	adUnit2.Sizes = []openrtb.Format{
-		{
-			W: 8,
-			H: 10,
-		},
-	}
-	adUnit2.Params = json.RawMessage("{\"siteId\":\"1111\"}")
-
-	pbBidder := pbs.PBSBidder{
-		BidderCode: "bannerCode",
-		AdUnits: []pbs.PBSAdUnit{
-			adUnit1,
-			adUnit2,
-		},
-	}
-	bids, err := an.Call(ctx, &pbReq, &pbBidder)
-	if err != nil {
-		t.Fatalf("Should not have gotten an error: %v", err)
-	}
-
-	if len(bids) != 2 {
-		t.Fatalf("Should have received two bid")
-	}
-
-	bid := findBidByAdUnitCode(bids, adUnit1.Code)
-	if adUnit1.Sizes[0].H != bid.Height || adUnit1.Sizes[0].W != bid.Width {
-		t.Fatalf("Received the wrong size")
-	}
-
-	bid = findBidByAdUnitCode(bids, adUnit2.Code)
-	if adUnit2.Sizes[0].H != bid.Height || adUnit2.Sizes[0].W != bid.Width {
-		t.Fatalf("Received the wrong size")
-	}
-}
-
-func TestIxTwoSlotMultiSizeOnlyValidIXSizeResponse(t *testing.T) {
-
-	server := httptest.NewServer(
-		http.HandlerFunc(dummyIXServer),
-	)
-	defer server.Close()
-
-	conf := *adapters.DefaultHTTPAdapterConfig
-	an := NewIxAdapter(&conf, server.URL)
-	ctx := context.TODO()
-	pbReq := pbs.PBSRequest{}
-	adUnit := getAdUnit()
-	adUnit.Sizes = append(adUnit.Sizes, openrtb.Format{W: 20, H: 22})
-
-	pbBidder := pbs.PBSBidder{
-		BidderCode: "bannerCode",
-		AdUnits: []pbs.PBSAdUnit{
-			adUnit,
-		},
-	}
-	bids, err := an.Call(ctx, &pbReq, &pbBidder)
-
-	if err != nil {
-		t.Fatalf("Should not have gotten an error: %v", err)
-	}
-
-	if len(bids) != 2 {
-		t.Fatalf("Should have received 2 bids")
-	}
-
-	for _, size := range adUnit.Sizes {
-		if !bidResponseForSizeExist(bids, size.H, size.W) {
-			t.Fatalf("Missing bid for specified size %d and %d", size.W, size.H)
-		}
-	}
-}
-
-func bidResponseForSizeExist(bids pbs.PBSBidSlice, h uint64, w uint64) bool {
-	for _, v := range bids {
-		if v.Height == h && v.Width == w {
-			return true
-		}
-	}
-	return false
-}
-
-func findBidByAdUnitCode(bids pbs.PBSBidSlice, c string) *pbs.PBSBid {
-	for _, v := range bids {
-		if v.AdUnitCode == c {
-			return v
-		}
-	}
-	return &pbs.PBSBid{}
-}
-
-func TestIxRequestLimit(t *testing.T) {
-
-	server := httptest.NewServer(
-		http.HandlerFunc(dummyIXServer),
-	)
-	defer server.Close()
-
-	conf := *adapters.DefaultHTTPAdapterConfig
-	an := NewIxAdapter(&conf, server.URL)
-	ctx := context.TODO()
-	pbReq := pbs.PBSRequest{}
-	adUnits := []pbs.PBSAdUnit{}
-
-	for i := 0; i < requestLimit+1; i++ {
-		adUnits = append(adUnits, getAdUnit())
-	}
-
-	pbBidder := pbs.PBSBidder{
-		BidderCode: "bannerCode",
-		AdUnits:    adUnits,
-	}
-
-	bids, err := an.Call(ctx, &pbReq, &pbBidder)
-
-	if err != nil {
-		t.Fatalf("Should not have gotten an error: %v", err)
-	}
-
-	if len(bids) != requestLimit {
-		t.Fatalf("Should have received %d bid", requestLimit)
-	}
+	maxRequests = 2
+	adapterstest.RunJSONBidderTest(t, "ixtest", NewIxBidder(nil, "http://ib.adnxs.com/openrtb2"))
```

Comment:
Please use a more obvious test url, such as `http://fakehost/openrtb2` or the like.